### PR TITLE
[fix] warning for creating stock entry after sales invoice (causing missing gl entries)

### PR DESCRIPTION
**Pull request by RobertSchouten**
gives warnings and fix's for issue #5945

### CODE REVIEW

File: erpnext/stock/doctype/stock_entry/stock_entry.py [hunk 1]
```diff
@@ -65,6 +66,38 @@ def on_cancel(self):
...
+	def validate_posting_date(self):
+		allow_negative_stock = cint(frappe.db.get_value("Stock Settings", None, "allow_negative_stock"))
+		items, warehouses = self.get_items_and_warehouses()
+		previous_stock_vouchers = self.get_previous_stock_vouchers(self.posting_date, self.posting_time, warehouses,
+		                                                           items)
+
+		if allow_negative_stock and previous_stock_vouchers:
+			for voucher_type, voucher_no in previous_stock_vouchers:
+				frappe.msgprint(
+					_("The posting date is after {0}: {1}, this will cause missing data in the General Ledger").format(
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_entry/stock_entry.py]: Why this validation is required?

File: erpnext/stock/doctype/stock_entry/stock_entry.py [hunk 2]
```diff
@@ -65,6 +66,38 @@ def on_cancel(self):
...
+		condition = ""
+
+		if for_items:
+			condition += " and item_code in ({})".format(", ".join(["%s"] * len(for_items)))
+			values += for_items
+
+		for d in frappe.db.sql("""select distinct sle.voucher_type, sle.voucher_no
+			from `tabStock Ledger Entry` sle
+			where timestamp(sle.posting_date, sle.posting_time) <= timestamp(%s, %s) {condition}
+			and warehouse = "Finished Goods - IAG"
```
Comment (nabinhait) [inline erpnext/stock/doctype/stock_entry/stock_entry.py]: hardcoded warehouse?

Comment (RobertSchouten) [inline erpnext/stock/doctype/stock_entry/stock_entry.py]: appologies am commiting others work will re-evaluate

File: erpnext/stock/doctype/stock_entry/stock_entry.py (hunk 1 above)
Comment (RobertSchouten) [inline erpnext/stock/doctype/stock_entry/stock_entry.py]: the validation prevents stock entries being posted after there relevant sales invoice's so when the stock entry is updated the sales invoice's gl entries are also updated

**Comment by nabinhait** [comment]
Thanks for the PR. Removed posting date validation and merged, did not understand the reason for that.

**Pull request merged**